File: src/contexts/chat/infrastructure/mock/in-memory-scheduler.gateway.ts
import { ISchedulerGateway } from '@contexts/chat/application/ports/scheduler.gateway';
import { JobAggregate } from '@contexts/chat/domain/aggregates/job.aggregate';
import { Injectable } from '@nestjs/common';

import { InMemoryNodesGateway } from './in-memory-nodes.gateway';

/**
 * Stands in for nexora-scheduler. In v0, with a single node registered,
 * there's no real decision to make: it's a direct router.
 *
 * Depends on the concrete InMemoryNodesGateway (not the INodesGateway
 * port) so Nest's DI wires the same relationship the Go/Java v0s built
 * by hand — onlyNodeId() isn't part of the port's public contract.
 */
@Injectable()
export class InMemorySchedulerGateway implements ISchedulerGateway {
  constructor(private readonly nodesGateway: InMemoryNodesGateway) {}

  async selectNode(_job: JobAggregate): Promise<string> {
    return this.nodesGateway.onlyNodeId();
  }
}

File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/infrastructure/mock/InMemorySchedulerGateway.java
package com.sisqueslabs.nexora.api.contexts.chat.infrastructure.mock;

import org.springframework.stereotype.Component;

import com.sisqueslabs.nexora.api.contexts.chat.application.port.SchedulerGateway;
import com.sisqueslabs.nexora.api.contexts.chat.domain.entities.Job;

/**
 * Stands in for nexora-scheduler. In v0, with a single node registered,
 * there's no real decision to make: it's a direct router.
 *
 * Depends on {@link InMemoryNodesGateway} (rather than a plain node ID
 * value) so Spring's dependency injection wires the same relationship
 * the Go v0 built by hand in chat.go's Register function.
 */
@Component
public class InMemorySchedulerGateway implements SchedulerGateway {

    private final InMemoryNodesGateway nodesGateway;

    public InMemorySchedulerGateway(InMemoryNodesGateway nodesGateway) {
        this.nodesGateway = nodesGateway;
    }

    @Override
    public String selectNode(Job job) {
        return nodesGateway.onlyNodeId();
    }
}
